src/cdk/tree/tree-data.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Observable} from 'rxjs/Observable';
+
+/** Interface for data in flat tree. Tree users should implement this interface to use CdkTree. */
+export interface FlatNode {
+  level: number;
```

Comment:
Should we do a `getLevel` method instead of a property? It might be annoying if someone has a `level` property in their data that means something else. (Same for `expandable` -> `isExpandable`). 

I also wonder if there's a way to could express this in the template (similar to `when`) instead of requiring an interface, but I forget if we already explored that option

Reply:
We didn't explore that option. 

I can try to express it in the template, but `TreeControl` is also using that to expand/collapse subtrees/the whole tree.

If we want to express this in the template, we can also pass that info to `TreeControl`, or put `getLevel` `getChildren` and `isExpandable` functions in `TreeControl` and let other components use functions in `TreeControl`.

Reply:
Now moved `getLevel` `isExpandable` and `getChildren` to `TreeControl`. Removed `NestedNode` and `FlatNode` types and users can use any type of data.